kafka/producer-manager.go

```diff
@@ -0,0 +1,144 @@
+package kafka
+
+import (
+	"sync"
+
+	"time"
+
+	"net/http"
+
+	"github.com/Shopify/sarama"
+	"github.com/Sirupsen/logrus"
+	"github.com/pressly/chi"
+	"github.com/thrawn01/args"
+	"golang.org/x/net/context"
+)
+
+type contextKey int
+
+const (
+	producerManagerKey contextKey = 1
+)
+
+func SetProducerManager(ctx context.Context, manager *ProducerManager) context.Context {
+	return context.WithValue(ctx, producerManagerKey, manager)
+}
+
+func GetProducerManager(ctx context.Context) *ProducerManager {
+	obj, ok := ctx.Value(producerManagerKey).(*ProducerManager)
+	if !ok {
+		panic("No kafka.ProducerManager found in context")
+	}
+	return obj
+}
+
+type ProducerManager struct {
+	done      chan struct{}
+	parser    *args.ArgParser
+	current   chan Producer
```

Comment:
I like the ``self.session`` solution mainly because for me it's easier to understand and because it's shorter / cleaner. The channels felt a bit exotic for this use case. Nevertheless I think it was very elegant! Great job using channels!

No need in the extra PR unless you want to try it out for your own interest. The benchmark script captures two concepts pretty well :+1: